kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -225,12 +235,61 @@ int size() {
     return buffer.size();
   }
 
-  public void flush(long offset) {
-    if (buffer.size() < flushThreshold) {
-      LOG.debug("Ignoring flush() - commit buffer {} smaller than flush threshold {}",
+  private long totalBytesBuffered() {
+    return buffer.values().stream()
+        .mapToLong(e -> plugin.bytes(e.key).get().length + (e.isTombstone ? 0 : e.value.length))
+        .sum();
+  }
+
+  private boolean shouldFlush() {
+    boolean recordsTrigger = false;
+    boolean bytesTrigger = false;
+    boolean timeTrigger = false;
+    if (buffer.size() >= flushTriggers.getRecords()) {
+      LOG.info("will flush due to records buffered {} over trigger {}",
           buffer.size(),
-          flushThreshold
+          flushTriggers.getRecords()
+      );
+      recordsTrigger = true;
+    } else {
+      LOG.debug("records buffered {} not over trigger {}",
+          buffer.size(),
+          flushTriggers.getRecords()
+      );
+    }
+    // This is a little inefficient (as opposed to maintaining the size as we go along).
+    // On the other hand, it's much less error-prone.
```

Comment:
let's not call this if the configuration is set to `Long.MAX_VALUE`

Reply:
I got a little nervous about this so I just refactored the buffer part into a size-tracking buffer